compiler/statements.go

```diff
@@ -39,10 +44,29 @@ func DoMetaCommand(buffer InputBuffer) MetaCommandResult {
 
 func PrepareStatement(buffer InputBuffer, statement *Statement) PrepareResult {
 	if len(buffer.Buffer) > 6 {
-		if buffer.Buffer[:6] == "insert" {
-		statement.Type = StatementInsert
-		return PrepareSuccess
-	}
+		// bufferArguments contains a slice of arguments in buffer.Buffer separated by a space
+		bufferArguments := strings.Fields(buffer.Buffer)
+		if bufferArguments[0] == "insert" {
+			statement.Type = StatementInsert
+			if len(bufferArguments) < 4 {
+				return PrepareSyntaxError
+			} else {
+				i, err := strconv.Atoi(bufferArguments[1])
+				if err != nil {
+					log.Printf("%q is not a valid id\n", bufferArguments[1])
+					return PrepareSyntaxError
+				} else {
+					statement.rowToInsert.id = int32(i)
+				}
+				statement.rowToInsert.username = bufferArguments[2]
+				statement.rowToInsert.username = bufferArguments[3]
+			}
+			return PrepareSuccess
+		}
+		// if buffer.Buffer[:6] == "insert" {
```

Comment:
Can we make use of Golang's formatter? `go fmt ./...` <= run this in root to properly format the codebase before creating PRs :)